### PR TITLE
Add timer packets to demo playback

### PR DESCRIPTION
**Pull request by Crystalwarrior**
Add an "active" check for timers so we can know if the timer is active or not
Add timer packets to the demo file
Emit a skip_timers signal that will adjust the timers to skip the difference when max_wait kicks in
Stop static-casting qint64 to int when we _want_ it to be qint64 for timers
Add a new skip_clocks functionality
Set up demoserver with a signal that asks the courtroom to skip some msecs in timers

### CODE REVIEW

File: src/courtroom.cpp [hunk 1]
```diff
@@ -5446,15 +5446,26 @@ void Courtroom::start_clock(int id, qint64 msecs)
...
 void Courtroom::set_clock(int id, qint64 msecs)
 {
   if (id >= 0 && id < max_clocks && ui_clock[id] != nullptr)
   {
-    ui_clock[id]->set(static_cast<int>(msecs), true);
+    ui_clock[id]->set(msecs, true);
+  }
+}
+
+// Used by demo playback to adjust for max_wait skpis
```
Comment (in1tiate) [inline src/courtroom.cpp:5461]:
```suggestion
// Used by demo playback to adjust for max_wait skips
```

**Review by in1tiate — COMMENTED** [review]
looks good to me, i'll leave it to @oldmud0 to approve though

File: src/aoapplication.cpp [hunk 2]
```diff
@@ -171,6 +173,12 @@ void AOApplication::ms_connect_finished(bool connected, bool will_retry)
   }
 }
 
+void AOApplication::skip_timers(qint64 msecs)
+{
+  if (courtroom_constructed)
+    w_courtroom->skip_clocks(msecs);
+}
+
```
Comment (oldmud0) [inline src/aoapplication.cpp:181]: Why not connect skip_timers directly from the demo server to the courtroom when AOApplication is constructing the courtroom?

**Comment by oldmud0** [comment]
Also, I told you not to make those behemoth commits...

**Comment by Crystalwarrior** [comment]
> 
> 
> Also, I told you not to make those behemoth commits... One feature/fix/change per commit please. At most two.

Fixed it 😳

**Review by oldmud0 — CHANGES_REQUESTED** [review]
Build error: `error C2065: 'min_wait': undeclared identifier`

**Review by oldmud0 — APPROVED** [review]

**Pull request merged**